nighthawk/source/server/http_test_server_filter.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+package nighthawk.server;
+
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+
+// TODO(oschaaf): HeaderValue(Option) are copied from 
+// import "envoy/api/v2/core/base.proto";
+// Unfortunately, visibility aspects seem to prevent re-use as of writing this.
+
+// Header name/value pair.
+message HeaderValue {
+  // Header name.
+  string key = 1 [(validate.rules).string = {min_bytes: 1, max_bytes: 16384}];
+
+  // Header value.
+  //
+  // The same :ref:`format specifier <config_access_log_format>` as used for
+  // :ref:`HTTP access logging <config_access_log>` applies here, however
+  // unknown header values are replaced with the empty string instead of `-`.
+  string value = 2 [(validate.rules).string.max_bytes = 16384];
+}
+
+// Header name/value pair plus option to control append behavior.
+message HeaderValueOption {
+  // Header name/value pair that this option applies to.
+  HeaderValue header = 1 [(validate.rules).message.required = true];
+
+  // Should the value be appended? If true (default), the value is appended to
+  // existing values.
+  google.protobuf.BoolValue append = 2;
+}
+
+message TestServer {
```

Comment:
I would probably rename this as `ResponseOptions`. Should this proto go somewhere outside of source, e.g. into `nighthawk/proto` or `nighthawk/api/server`? You'll be building up a number of protos that define the NH APIs in the future.

Reply:
Yes, that makes sense, done. Also added moving the other proto's out to the backlog.